### PR TITLE
Update CHANGELOG for 10.0.0

### PR DESCRIPTION
**Pull request by onurctirtir**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,125 @@
...
+
+* Removes `upgrade_to_reference_table` UDF
+
+* Renames 'master' to 'distributed' for columns of `citus_dist_stat_activity`
+
+* Renames 'master' to 'distributed' for columns of `citus_worker_stat_activity`
+
+* Avoids establishing new connections until the previous ones are finalized
+
+* merge cstore into the citus repo
```
Comment (onurctirtir) [inline CHANGELOG.md:83]: I think we should re-phrase columnar entry and move it to top, maybe `Add columnar tables` or something ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,125 @@
+### citus v10.0 (February 16, 2021) ###
+
+* Adds shard rebalancer
```
Comment (onurctirtir) [inline CHANGELOG.md:3]: better wording ?

**Review by onurctirtir — COMMENTED** [review]
I will shorten long entries soon

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,139 @@
...
+
+* Removes `citus.max_tracked_tasks_per_node` GUC
+
+* Removes `citus.sslmode` GUC
+
+* Removes `citus.task_tracker_delay` GUC
+
+* Removes the word 'master' from Citus UDFs
+
+* Removes unused UDF `mark_tables_colocate`
```
Comment (onderkalaci) [inline CHANGELOG.md:82]: I guess it was `mark_tables_colocated` ?

File: CHANGELOG.md (hunk 1 above)
Comment (onderkalaci) [reply CHANGELOG.md:92]: Yes, sure we should do that. This should be the first item in the release notes. Maybe, use the sentence from the wiki: https://github.com/citusdata/citus/tree/master/src/backend/columnar#introduction
```Adds support for per-table option for columnar storage```

File: CHANGELOG.md (hunk 2 above)
Comment (onurctirtir) [reply CHANGELOG.md:3]: maybe: `Adds community edition of shard rebalancer with `rebalance_table_shards` & `citus_drain_node` UDFs`

File: CHANGELOG.md (hunk 1 above)
Comment (onurctirtir) [reply CHANGELOG.md:92]: maybe `Adds columnar tables with columnar access method`

File: CHANGELOG.md (hunk 2 above)
Comment (onderkalaci) [reply CHANGELOG.md:3]: Maybe two items:
- Adds rebalance_table_shards() to rebalance the shards across the nodes
- Adds citus_drain_node() to move all shards away from any node

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,139 @@
+### citus v10.0 (February 16, 2021) ###
+
```
Comment (onderkalaci) [inline CHANGELOG.md:2]: I think we should have the most interesting things at the top:
1. columnar
2. rebalancer
3. Enables Single-node Citus for production workloads (add this as an item as it is not explicitly mentioned in others)
4. Local table distributed table joins
5. Foreign key to/from local tables

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,139 @@
...
+* Introduces adaptive connection management for local nodes
+
+* Adds `pg_catalog.citus_shards`, `pg_catalog.citus_shards_on_worker` and
+  `pg_catalog.citus_shard_indexes_on_worker` views
+
+* Adds `pg_catalog.time_partitions` view for simple (time) partitions and their
+  access methods
+
+* Adds `public.citus_tables` view
```
Comment (onderkalaci) [inline CHANGELOG.md:30]: I'd put this item above all other views in the list

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,139 @@
...
+
+* Propagates `ALTER SCHEMA RENAME` statements
+
+* Propagates `CREATE STATISTICS` statements across workers.
+
+* Propagates `DROP STATISTICS` statements across the workers
+
+* Propagates all types of `ALTER STATISTICS` statements.
+
+* Propagates columnar table settings for distributed tables
```
Comment (onderkalaci) [inline CHANGELOG.md:58]: I'm not sure we need `Propagates columnar table settings for distributed tables` as that is part of columnar anyway?

Comment (onurctirtir) [reply CHANGELOG.md:64]: yeah, I forgot to ask but I agree that we should remove that

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,139 @@
...
+* Adds `public.citus_tables` view
+
+* Adds `remove_local_tables_from_metadata` UDF for local tables automatically
+  added to metadata when defining foreign keys with reference tables
+
+* Adds support for `CREATE INDEX` commands without index name on citus tables
+
+* Adds support for `CREATE TABLE ... USING table_access_method` statements
+
+* Adds support for co-located/recurring correlated subqueries
```
Comment (onderkalaci) [inline CHANGELOG.md:45]: These two items `colocated/recurring subqueries` are significant enough. Could come right after the top list that I suggested above

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,141 @@
...
+  access methods
+
+* Adds `remove_local_tables_from_metadata` UDF for local tables automatically
+  added to metadata when defining foreign keys with reference tables
+
+* Adds support for `CREATE INDEX` commands without index name on citus tables
+
+* Adds support for `CREATE TABLE ... USING table_access_method` statements
+
+* Adds support for co-located/recurring sublinks in the target list
```
Comment (onderkalaci) [inline CHANGELOG.md:47]: We had two items regarding `colocated/recurring subqueries`. The other seems gone after your last commit?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,141 @@
...
+* Fixes handling indexes when adding local table to metadata
+
+* Fixes `undistribute_table` when table has column with `GENERATED ALWAYS AS
+  STORED` expressions
+
+* Improves concurrent index creation failure message
+
+* Prevents empty placement creation in the coordinator when executing
+  `master_create_empty_shard()`
+
```
Comment (onurctirtir) [inline CHANGELOG.md:138]: Should we have this pr in changelog https://github.com/citusdata/citus/pull/4489 ?

(I explicitly didn't add a description for that as we are just exposing adding local tables to metadata)

File: CHANGELOG.md (hunk 8 above)
Comment (onurctirtir) [reply CHANGELOG.md:47]: actually just moved it to an upper level 4794b9a

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,141 @@
...
+* Adds cascade option to `undistribute_table` UDF for foreign key connected
+  relations
+
+* Allows distributed table creation immediately after CREATE EXTENSION citus
+
+* Automatically adds coordinator to the metadata before the first distributed
+  table created
+
+* Introduces adaptive connection management for local nodes
```
Comment (onderkalaci) [inline CHANGELOG.md:30]: Minor: can we get local node related things close to each other, most of them are already here, just move few of the others here:
- Allows distributed table creation immediately after CREATE EXTENSION citus
- Automatically adds coordinator to the metadata before the first distributed table created
- Introduces adaptive connection management for local nodes
- Adds support for local execution in INSERT..SELECT commands that perform re-partitioning
- Adds support for local execution in INSERT..SELECT commands that perform re-partitionin

File: CHANGELOG.md (hunk 8 above)
Comment (onurctirtir) [reply CHANGELOG.md:47]: (I should move the other one near that)

Comment (onderkalaci) [reply CHANGELOG.md:47]: Alright, I think we should move this just next to it

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,141 @@
...
+
+* Removes unused UDF `mark_tables_colocated`
+
+* Removes `upgrade_to_reference_table` UDF
+
+* Renames 'master' to 'distributed' for columns of `citus_dist_stat_activity`
+
+* Renames 'master' to 'distributed' for columns of `citus_worker_stat_activity`
+
+* Avoids establishing new connections until the previous ones are finalized
```
Comment (onderkalaci) [inline CHANGELOG.md:94]: These two ` Avoids establishing ..` and ` Avoids re-using connections for in ..` are actually significant as well. Shall we move these to just before ` Adds support for `CREATE INDEX ..`

File: CHANGELOG.md (hunk 9 above)
Comment (onderkalaci) [reply CHANGELOG.md:138]: I think we should not as you suggest

File: CHANGELOG.md (hunk 11 above)
Comment (onurctirtir) [reply CHANGELOG.md:94]: makes sense

File: CHANGELOG.md (hunk 8 above)
Comment (onurctirtir) [reply CHANGELOG.md]: 71b3977913dff9db39a6d06c9d5b3ee46bda593b

File: CHANGELOG.md (hunk 10 above)
Comment (onurctirtir) [reply CHANGELOG.md:32]: c2aaed64db4937655de8ab7ee6cc317ec490ab4c

File: CHANGELOG.md (hunk 11 above)
Comment (onurctirtir) [reply CHANGELOG.md]: b37a3e6683c95c997a08a95495ee212388a94bed

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,141 @@
...
+
+* Adds support for per-table option for columnar storage
+
+* Adds `rebalance_table_shards` to rebalance the shards across the nodes
+
+* Adds `citus_drain_node` to move all shards away from any node
+
+* Enables single-node Citus for production workloads
+
+* Adds support for local distributed/subquery joins
```
Comment (onderkalaci) [inline CHANGELOG.md:11]: Maybe `Adds support for local distributed/subquery joins -> Adds support for local table and distributed table/subquery joins?`

the former sounds a little vague

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,141 @@
...
+
+* Removes `upgrade_to_reference_table` UDF
+
+* Renames 'master' to 'distributed' for columns of `citus_dist_stat_activity`
+
+* Renames 'master' to 'distributed' for columns of `citus_worker_stat_activity`
+
+* Chooses the default co-location group deterministically
+
+* Deletes transactions when removing a node
```
Comment (onderkalaci) [inline CHANGELOG.md:100]: Maybe `Deletes transactions when removing a node` -> `Deletes distributed transaction entries when removing a node`

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,141 @@
...
+
+* Propagates all types of `ALTER STATISTICS` statements.
+
+* Avoids establishing new connections until the previous ones are finalized
+
+* Avoids re-using connections for intermediate results
+
+* Improves performance in some subquery pushdown cases
+
+* Removes `citus.binary_master_copy_format` GUC
```
Comment (onderkalaci) [inline CHANGELOG.md:70]: Should we change these to `Removes unused ....`? Because these were all deprecated before Citus 10

**Review by onderkalaci — APPROVED** [review]
What a powerful release!

File: CHANGELOG.md (hunk 14 above)
Comment (onurctirtir) [reply CHANGELOG.md:70]: makes sense

**Pull request merged**